.github/workflows/test.yml

```diff
@@ -0,0 +1,79 @@
+name: Test
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+
+env:
+  CI: "true"
+  FORCE_COLOR: "true"
+
+jobs:
+  test:
+    name: Node@${{ matrix.node }} pup@${{ matrix.puppeteer_version }} on ${{ matrix.os }}
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        puppeteer_version:
+          - 5.0.0
+          - 2.1.1 # Chromium 79.0.3942.0, Oct 24 2019
+          # - 2.0.0 # Chromium 79.0.3942.0, Oct 24 2019
+          # - 1.20.0 # Chromium 78.0.3882.0, Sep 13 2019
+          # - 1.15.0 # Chromium 75.0.3765.0, Apr 26 2019
+          # - 1.9.0 # Chromium 71.0.3563.0, Oct 4, 2018
+          # - 1.6.2 # Chromium 69.0.3494.0, Aug 1, 2018
+        node:
+          - 10
+          - 12
+          - 14
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          # - windows-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: 'Fix for: error fsevents@2.1.2: The platform "linux" is incompatible with this module.'
+        run: npx json -I -f package.json -e 'this.resolutions={}'
+
+      - name: yarn install
+        uses: bahmutov/npm-install@v1
+
+      - name: yarn bootstrap
+        run: yarn bootstrap
+
+      - name: install puppeteer@${{ matrix.puppeteer_version }}
+        run: yarn lerna add --dev puppeteer@${{ matrix.puppeteer_version }}
+
+      - name: lerna link
+        run: yarn lerna link
+
+      - name: lerna build
+        run: yarn lerna run build --concurrency 1
+
+      - name: debug
+        run: |
+          yarn list --pattern "puppeteer|puppeteer-extra"
+          file node_modules/puppeteer-extra/dist/index.cjs.js
+
+      - name: test
+        run: yarn test-ci
+
+      - name: test with xvfb
```

Comment:
Oh, I think we're effectively running the tests twice now? It should be sufficient to only once run the tests with xvfb support (in case there are headful tests)

Reply:
Yeah, I am not too familiar with this. I figured one way runs them headless and the other will run them headed?

Or is it not the case?

Reply:
Do we want to test in headed mode only then? (with xvfb on)

Reply:
Do we want to test in headed mode only then? (with xvfb on)

Yup

Reply:
Ok, done.

But you don't think there might be some differences between headed and headless mode? There are no detections that check for headless?

Reply:
> But you don't think there might be some differences between headed and headless mode? There are no detections that check for headless?

Not that I recall. :-) Headless tests will run normally even in the presence of a virtual frame buffer which is needed for headful tests.